eng/common/pipelines/templates/jobs/sdk-generation.yml

```diff
@@ -0,0 +1,160 @@
+        fi
+      displayName: 'Set SpecRepoCommit variable'
+
+    - task: NodeTool@0
+      inputs:
+        versionSpec: '22.x'
+      displayName: 'Install Node.js'
+
+    - script: |
+        npm install -g @azure-tools/spec-gen-sdk
+      displayName: 'Install spec-gen-sdk'
+
+    - script: |
+        optional_params=""
+        sdk_gen_info="sdk generation from Config : "
+
+        if [ "${{ parameters.ConfigType }}" = "TypeSpec" ]; then
+          optional_params="$optional_params --tsp-config-relative-path ${{ parameters.TspConfigPath }}"
+          sdk_gen_info="$sdk_gen_info '${{ parameters.TspConfigPath }}',"
+        else
+          optional_params="$optional_params --readme-relative-path ${{ parameters.ReadmePath }}"
+          sdk_gen_info="$sdk_gen_info '${{ parameters.ReadmePath }}',"
+        fi
+        if [ "${{ parameters.ApiVersion }}" != "not-specified" ]; then
+          optional_params="$optional_params --api-version ${{ parameters.ApiVersion }}"
+          sdk_gen_info="$sdk_gen_info API Version: '${{ parameters.ApiVersion }}',"
+        fi
+
+        sdk_gen_info="$sdk_gen_info and CommitSHA: '$(SpecRepoCommit)'"
+        echo "##vso[task.setvariable variable=GeneratedSDKInformation]$sdk_gen_info"
+        echo "Generated SDK Information : $sdk_gen_info"
+
+        spec-gen-sdk \
+          --scp "$(SpecRepoDirectory)" \
+          --sdp "$(SdkRepoDirectory)" \
+          --wf "$(System.DefaultWorkingDirectory)" \
+          -l "${{ parameters.SdkRepoNameForSpecGen }}" \
+          -c "$(SpecRepoCommit)" \
+          $optional_params
+      displayName: 'Generate SDK'
+
+    - ${{ if eq(parameters.CodePushRequired, true) }}:
+      - template: /eng/common/pipelines/templates/steps/git-push-changes.yml
+        parameters:
+          BaseRepoBranch: sdk-generation-from-$(Build.DefinitionName)_$(Build.BuildId)
+          BaseRepoOwner: azure-sdk
+          CommitMsg: $(GeneratedSDKInformation)
+          TargetRepoOwner: ${{ parameters.SdkRepoOwner }}
+          TargetRepoName: ${{ parameters.SdkRepoName }}
+          PushArgs: "--force"
+          WorkingDirectory: $(SdkRepoDirectory)
+          ScriptDirectory: $(SdkRepoDirectory)/eng/common/scripts
+
+    - task: PowerShell@2
+      displayName: Create pull request
+      condition: and(succeeded(), eq('${{ parameters.SdkPrRequired }}', true), eq(variables['HasChanges'], 'true'))
+      inputs:
+        pwsh: true
+        workingDirectory: $(SdkRepoDirectory)
+        filePath: $(SdkRepoDirectory)/eng/common/scripts/Submit-PullRequest.ps1
+        arguments: >
+          -RepoOwner "${{ parameters.SdkRepoOwner }}"
+          -RepoName "${{ parameters.SdkRepoName }}"
+          -BaseBranch "main"
+          -PROwner "azure-sdk"
+          -PRBranch "sdk-generation-from-$(Build.DefinitionName)_$(Build.BuildId)"
+          -AuthToken "$(azuresdk-github-pat)"
+          -PRTitle "sdk-generation-from-$(Build.DefinitionName)_$(Build.BuildId)"
```

Comment:
Output variables will be written to pipeline by the `spec-gen-sdk` to be consumed as PR title and PR body.